core/actions/data_preparation.ts

```diff
@@ -108,107 +114,168 @@ export class DataPreparation extends ActionBuilder<dataform.DataPreparation> {
       VerifyProtoErrorBehaviour.SUGGEST_REPORTING_TO_DATAFORM_TEAM
     );
   }
-}
 
-function parseDataPreparationDefinitionJson(dataPreparationAsJson: {
-  [key: string]: unknown;
-}): dataform.dataprep.DataPreparation {
-  try {
-    return dataform.dataprep.DataPreparation.create(
-      verifyObjectMatchesProto(
-        dataform.dataprep.DataPreparation,
-        dataPreparationAsJson as {
-          [key: string]: any;
-        },
-        VerifyProtoErrorBehaviour.SHOW_DOCS_LINK
-      )
-    );
-  } catch (e) {
-    if (e instanceof ReferenceError) {
-      throw ReferenceError(`Data Preparation parsing error: ${e.message}`);
+  private parseDataPreparationDefinitionJson(dataPreparationAsJson: {
```

Comment:
To match the naming convention of other action handlers, `verifyDataPreparationDefinition` would be better